### ISSUE TITLE
Search Results List Checkbox does not have label

### ISSUE THREAD

**Issue by Safei-Ashraf** [mozilla/perfcompare#303]
### Description

The accessibility test fails on the rendered list for search results, check boxes rendered on the left do not have appropriate labels and ARIA attributes

![image](https://user-images.githubusercontent.com/44810632/223384293-554bee81-5a65-4adf-85ec-50e405eebb17.png)

**command** to run the test locally `npm run test:a11y || true`

**Path to failing test:**  `src\__tests__\accessiblity\accessibility.test.tsx`
**Path to search components:**  `src\components\Search`

View more failing accessibility tests [here ](https://app.circleci.com/pipelines/github/mozilla/perfcompare/1103/workflows/8aaaeeb7-a7d5-422a-90fa-29257fe7c1e4/jobs/1107/parallel-runs/0/steps/0-107)

**Comment by cherish2003** [comment]
@kimberlythegeek i am outreachy applicant i want to work on this issue can you assign me

**Comment by Phyya** [comment]
@kimberlythegeek Please assign this to me

**Comment by taofeeqhamzat** [comment]
Hi @Carla-Moz I submitted a PR with a fix (#305) for this issue which actually duplicates #302.
Could you please take a look? I'd appreciate any feedback.

**Comment by taofeeqhamzat** [comment]
@Safei-Ashraf we seem to be working towards the same goal. How can we make it not confusing?

**Comment by Safei-Ashraf** [comment]
@taofeeqhamzat  it's okay for multiple interns to be working on the same issue, this will help evaluate different solutions/approaches and the interns' work in general.

**Comment by taofeeqhamzat** [comment]
> @taofeeqhamzat it's okay for multiple interns to be working on the same issue, this will help evaluate different solutions/approaches and the interns' work in general.

Correct. I think I get it now. When @ kimberlythegeek  said we could freely work on issues without waiting to be assigned, it wasn't immediately clear to me that that meant it was also okay for multiple interns to work on the same issue and submit multiple solutions, but I re-read [this contribution guide](https://docs.google.com/document/d/1NLKAR5B2Rc80lRUYWzU2ISrQCQhR4VsW5Qkz1hxE5n8/) and it clarified things. Some things may have changed since I last checked.

Got your response in chat.
